src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,147 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
+    private static class Node<T> {
+        T value;
+        Node<T> prev;
+        Node<T> next;
+
+        private Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (size == 0) {
+            Node<T> newNode = new Node<>(null, value, null);
+            first = newNode;
+            last = newNode;
+        } else {
+            Node<T> newNode = new Node<>(last, value, null);
+            last.next = newNode;
+            last = newNode;
+        }
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size == index) {
+            add(value);
+            return;
+        }
+        Node<T> targetNode = findNode(index);
+        Node<T> newNode = new Node(targetNode.prev, value, targetNode);
+        if (targetNode.prev == null) {
+            targetNode.prev = newNode;
+            first = newNode;
+        } else {
+            targetNode.prev.next = newNode;
+            targetNode.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T value : list) {
+            if (!add(value)) {
+                return false;
+            }
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return findNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> targetNode = findNode(index);
+        T removedValue = targetNode.value;
+        targetNode.value = value;
+        return removedValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> targetNode = findNode(index);
+        unlink(targetNode);
+        size--;
+        return targetNode.value;
     }
 
     @Override
     public boolean remove(T t) {
+        int index = 0;
+        Node<T> targetNode;
+        while (!(index >= size)) {
+            targetNode = findNode(index);
+            if (targetNode.value == null && t == null
+                    || targetNode.value.equals(t)) {
```

Comment:
```suggestion
            if (targetNode.value == null && t == null
                    || targetNode.value.equals(t)) {
```
what happens if targetNode.value will be null?